File: backend/models/Channel.js
const mongoose = require('mongoose');

const channelSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    trim: true
  },
  description: {
    type: String,
    default: ''
  },
  category: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Category',
    required: true
  },
  viewPermissions: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Role'
  }],
  postPermissions: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Role'
  }],
  order: {
    type: Number,
    default: 0
  },
  createdAt: {
    type: Date,
    default: Date.now
  }
}, {
  timestamps: true
});

module.exports = mongoose.model('Channel', channelSchema);

File: ECGCommunityApp/ECGCommunityApp/Models/Channel.swift
import Foundation

struct Category: Codable, Identifiable {
    let id: String
    let name: String
    let description: String
    let order: Int
    
    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case name, description, order
    }
}

struct Channel: Codable, Identifiable {
    let id: String
    let name: String
    let description: String
    let category: Category
    let viewPermissions: [Role]
    let postPermissions: [Role]
    let order: Int
    let canPost: Bool?
    
    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case name, description, category, viewPermissions, postPermissions, order, canPost
    }
}


